File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TaskListEditRequestEventArgs.vb
Imports System

Namespace ViewModels
    Public Class TaskListEditRequestEventArgs
        Inherits EventArgs

        Public Sub New(title As String, subtitle As String, saveAction As Action(Of String, String))
            Me.Title = title
            Me.Subtitle = subtitle
            Me.SaveAction = saveAction
        End Sub

        Public Property Title As String

        Public Property Subtitle As String

        Public ReadOnly Property SaveAction As Action(Of String, String)
    End Class
End Namespace

File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TaskListEditRequestEventArgs.cs
using System;

namespace TaskOTime.ViewModel.ViewModels
{
    public class TaskListEditRequestEventArgs : EventArgs
    {

        public TaskListEditRequestEventArgs(string title, string subtitle, Action<string, string> saveAction)
        {
            Title = title;
            Subtitle = subtitle;
            SaveAction = saveAction;
        }

        public string Title { get; set; }

        public string Subtitle { get; set; }

        public Action<string, string> SaveAction { get; private set; }
    }
}